gocd/bci_repo_publish.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/python3
+# (c) 2023 fvogt@suse.de
+# GPLv3-only
+
+# This is a "mini ttm" for the BCI repo. Differences:
+# * No :ToTest staging area
+# * Only the repo is built, so it needs BCI specific openQA queries
+# * The publishing location is arch specific
+# * Uses a token for releasing to the publishing project
+
+
+import cmdln
+import logging
+import ToolBase
+import requests
+import sys
+import time
+import re
+from lxml import etree as ET
+from openqa_client.client import OpenQA_Client
+from osc.core import http_GET, makeurl
+
+
+class BCIRepoPublisher(ToolBase.ToolBase):
+    def __init__(self):
+        ToolBase.ToolBase.__init__(self)
+        self.logger = logging.getLogger(__name__)
+        self.openqa = OpenQA_Client(server='https://openqa.suse.de')
+
+    def version_of_product(self, project, package, repo, arch):
+        """Get the build version of the given product build, based on the binary name."""
+        url = makeurl(self.apiurl, ['build', project, repo, arch, package])
+        root = ET.parse(http_GET(url)).getroot()
+        for binary in root.findall('binary'):
+            result = re.match(r'.*-Build(.*)-Media1.report', binary.get('filename'))
+            if result:
+                return result.group(1)
+
+        raise RuntimeError(f"Failed to get version of {project}/{package}")
+
+    def mtime_of_product(self, project, package, repo, arch):
+        """Get the build time stamp of the given product, based on _buildenv."""
+        url = makeurl(self.apiurl, ['build', project, repo, arch, package])
+        root = ET.parse(http_GET(url)).getroot()
+        mtime = root.xpath('/binarylist/binary[@filename = "_buildenv"]/@mtime')
+        return mtime[0]
+
+    def openqa_jobs_for_product(self, arch, version, build):
+        """Query openQA for all relevant jobs"""
+        values = {
+            'group': 'BCI repo',
+            'flavor': 'BCI-Repo-Updates',
```

Comment:
Currently this script is the configuration - if something changes we'll edit the script.

This will never work for anything other than the BCI repo anyway, so hardcoding everything is the easiest and most maintainable option.

Reply:
okay